flask/app/routes.py

```diff
@@ -517,3 +518,45 @@ def delete_tissue(id: int):
         return "Tissue has dataset(s), cannot delete", 422
     else:
         return "Not Found", 404
+
+
+@app.route("/api/families", methods=["POST"])
+@login_required
+def families():
+    if not request.json:
+        return "Request body must be JSON", 400
+
+    try:
+        updated_by = current_user.user_id
+        created_by = current_user.user_id
+    except:  # LOGIN_DISABLED
+        updated_by = 1
+        created_by = 1
+
+    fam_codename = request.json.get("family_codename")
+
+    # not sure if this is needed or if a different check would be better
+    if len(fam_codename) == 0:
+        return "Family codename must be at least 1 character long", 422
+
+    fam_query = models.Family.query.filter(
+        models.Family.family_codename == fam_codename
+    )
+
+    if fam_query.value("family_id"):
+        return "Family Codename already in use", 422
+
+    fam_objs = models.Family(
+        family_codename=fam_codename,
+        created_by=created_by,
+        updated_by=updated_by,
+    )
+
+    db.session.add(fam_objs)
+    transaction_or_abort(db.session.commit)
+
+    fam_id = fam_query.value("family_id")
```

Comment:
This value is already in the family object after the commit!

Reply:
oops good catch, completely forgot about that